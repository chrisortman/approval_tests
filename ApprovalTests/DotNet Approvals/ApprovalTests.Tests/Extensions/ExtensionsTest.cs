using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using ApprovalTests.Extensions;

namespace ApprovalTests.Tests.Extensions
{
	[TestFixture]
	public class ExtensionsTest
	{
		[Test]
		public void TestListToString()
		{
			string[] list = new[] { "one", "two", "three", "four" };

			Func<string,string> formatter = s => "{" +s+ "}";
			Approvals.Approve(list, formatter);
			list.ShouldBeApproved(formatter);
		}
		[Test]
		public void TestListSubobjects()
		{
			string[] list = new[] { "one", "two", "three", "four" };

			Func<string, string> formatter = s => "" + s.Length ;
			Approvals.Approve(list, formatter, "Number");
			list.ShouldBeApproved(formatter, "Number");
		}
	}
}
