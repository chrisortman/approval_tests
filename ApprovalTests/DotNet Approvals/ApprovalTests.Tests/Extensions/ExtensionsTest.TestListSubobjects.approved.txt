Number[0] = 3
Number[1] = 3
Number[2] = 5
Number[3] = 4
